packages/tools/client-debugger/client-debugger-view/src/DevtoolsView.tsx

```diff
@@ -177,15 +178,35 @@ export function DevtoolsView(): React.ReactElement {
 		// Query for supported feature set
 		messageRelay.postMessage(getSupportedFeaturesMessage);
 
+		// Add a timeout for the query
+		const timeoutId = setTimeout(() => {
+			if (!supportedFeatures) {
+				setQueryTimedOut(true);
+			}
+		}, queryTimeout);
+
 		return (): void => {
 			messageRelay.off("message", messageHandler);
+			clearTimeout(timeoutId); // Clear the timeout when cleaning up
```

Comment:
We probably want to add this to the message handler that receives the supported features message too. If we got a response to that, we don't need to let the timer expire.

Reply:
Make sense! Good caught up! Thx!